docs/content/concepts/graphql-indexer.mdx

```diff
@@ -354,6 +356,70 @@ The Consistent Store is a combined indexer and RPC service that is responsible f
+
+- **30 days** is a great baseline for dashboards and explorers that need recent activity and assets.
+- **90 days** improves support for longer-range pagination, historical lookups, or dApps with slower engagement cycles.
+
+You can configure your indexing pipelines to scope which data you include (such as events, objects, and transactions), and disable any components that aren’t needed.
+
+:::note
+
+Retaining long-term historical data in Postgres is not recommended unless required for specific apps.
+
+:::
+
+### Use the Archival Service and Store for historical lookups
+
+For all production deployments, you are strongly encouraged to pair Postgres with the [Archival Service](./archival-store.mdx) to support point lookups of transactions, objects, and checkpoints when relevant data does not exist in Postgres.
+
+- The Archival Service serves as the backend for historical versions and checkpoint data, reducing pressure on your Postgres instance.
+- While not strictly required, it is strongly recommended that you use the Archival Service in any production setup that aims to support `high-retention` GraphQL or gRPC workloads.
+
+Current implementation supports [GCP Bigtable](https://cloud.google.com/bigtable) which is a highly scalable and performant data store. If you plan to operate your own archival store, refer to `sui-kvstore` and `sui-kv-rpc` for indexer setup and RPC service implementation respectively. For the indexer setup, make sure to use the [custom indexing framework](../guides/developer/advanced/custom-indexer). If you're interested in contributing support for other scalable data stores, reach out on GitHub by creating a new issue.
+
+<details>
+<summary>
+`main.rs` in `sui-kvstore`
+</summary>
+<ImportContent source="crates/sui-kvstore/src/main.rs" mode="code" />
+</details>
+
+<details>
+<summary>
+`main.rs` in `sui-kv-rpc`
+</summary>
+<ImportContent source="crates/sui-kv-rpc/src/main.rs" mode="code" />
+</details>
+
+### Deployment strategies and trade-offs
+
+You don’t need to index everything to provide a reliable and performant GraphQL RPC service. In fact, many developers might need only the latest object and transaction data plus a few weeks to months of history. You can reduce operational overhead and improve query performance by:
+
+- Configuring a clear retention window (such as 30–90 days) in Postgres.
+- Using the Archival Service to handle deep historical queries, rather than retaining all versions in Postgres.
+
+When designing your deployment, consider the trade-offs between cost, reliability, and feature completeness:
+
+- Postgres-only with short-retention results in lower storage cost and faster performance, but limited historical coverage.
+- Postgres-only with high retention results in broader data coverage, but relatively higher storage cost and slower performance at scale.
+- Postgres with short-retention + Archival Service results in optimization for cost and completeness, ideal for production deployments.
```

Comment:
@abhinavg6 -- today it's practically quite difficult to run Postgres-only with short-retention (the system will behave strangely, e.g. pruning object contents that are live, but were created a long time ago), so the only usable options are to store everything in postgres, or to store KV data in the archival service.